warehouse/query-core/src/main/java/datawave/query/planner/DefaultQueryPlanner.java

```diff
@@ -809,6 +822,11 @@ protected ASTJexlScript updateQueryTree(ScannerFactory scannerFactory, MetadataH
             config.setQueryTree(timedEnforceUniqueDisjunctionsWithinExpressions(timers, config.getQueryTree()));
         }
 
+        // rewrite regex nodes, optimistically
+        if (rewriteRegexTerms) {
```

Comment:
Would probably rather see this after the RegexIndexExpansion visitor call.  If we can expand to explicit terms, we should.  Otherwise, rewrite the parts that can't if the values are not selective enough.